tools/mpy_ld.py

```diff
@@ -1207,13 +1215,27 @@ def link_objects(env, native_qstr_vals_len):
             sym.section = env.obj_table_section
         elif sym.name in env.known_syms:
             sym.resolved = env.known_syms[sym.name]
+        elif sym.name in env.externs:
+            # Fixed-address symbols do not need pre-processing.
+            continue
         else:
             if sym.name in fun_table:
                 sym.section = mp_fun_table_sec
                 sym.mp_fun_table_offset = fun_table[sym.name]
             else:
                 undef_errors.append("{}: undefined symbol: {}".format(sym.filename, sym.name))
 
+    seen_externs = set()
+    for sym in env.externs:
+        if sym in env.known_syms and sym not in seen_externs:
+            log(
+                LOG_LEVEL_1,
+                "Symbol {} is a fixed-address symbol at {:08x} and is also provided from an object file".format(
+                    sym, env.externs[sym]
+                ),
+            )
+            seen_externs.add(sym)
```

Comment:
Sorry to keep going back and forth here, but is `seen_externs` actually needed?  Because `env.externs` is guaranteed to have unique entries, you'll never see the same symbol twice.

Reply:
It might be a leftover of the previous loop iteration, see https://github.com/micropython/micropython/compare/2b46c83bab15c342cf16bc498f4b01a018a6acb9..23515be466d5c6e1f780e4057a1439e6aab1e58e where the `btree` module would report redundant symbols multiple times.

Looking back at it, yes, it should have been removed along with the old loop.  I'll fix this right away, thanks!